masonry_core/src/core/properties.rs

```diff
@@ -3,8 +3,17 @@
 
 use anymap3::{AnyMap, Entry};
 
-// TODO - Add PropertyValue wrapper struct that implements receiver trait.
-// Return PropertyValue<T> instead of Option<T> from methods.
+/// A marker trait that indicates that a type is intended to used as a widget's property.
+///
+/// See [properties documentation](crate::doc::doc_04b_widget_properties) for
+/// a full explanation of the general concept.
+///
+/// Note that if a type `Foobar` implements Property, that tells you that Foobar is meant
+/// to be a property of *some* widget, but it doesn't tell you *which* widget accepts Foobar
+/// as a property.
+/// That information is deliberately not encoded in the type system.
+/// We might change that in a future version.
+pub trait Property: Default + 'static {}
```

Comment:
Why `Default`? That superbound seems to be unused.

Reply:
Future changes.

Reply:
I don't think it's good practise to add it in this PR, but I won't block on it.